sqlmesh/dbt/target.py

```diff
@@ -860,6 +863,97 @@ def to_sqlmesh(self, **kwargs: t.Any) -> ConnectionConfig:
+      custom_settings: [{}] # A dictionary/mapping of custom ClickHouse settings for the connection - default is empty.
+      schema: [default] # ClickHouse database for dbt models, not used by SQLMesh
+      driver: [http] # http or native.  If not set this will be autodetermined based on port setting, not used by SQLMesh
+      retries: [1] # Number of times to retry a "retriable" database exception (such as a 503 'Service Unavailable' error), not used by SQLMesh
+      compression: [<empty string>] # Use gzip compression if truthy (http), or compression type for a native connection, not used by SQLMesh
+      cluster_mode: [False] # Use specific settings designed to improve operation on Replicated databases (recommended for ClickHouse Cloud), not used by SQLMesh
+      use_lw_deletes: [False] # Use the strategy `delete+insert` as the default incremental strategy, not used by SQLMesh
+      check_exchange: [True] # Validate that clickhouse support the atomic EXCHANGE TABLES command. Not used by SQLMesh.
+      local_suffix: [_local] # Table suffix of local tables on shards for distributed materializations, not used by SQLMesh
+      local_db_prefix: [<empty string>] # Database prefix of local tables on shards for distributed materializations, not used by SQLMesh
+      allow_automatic_deduplication: [False] # Enable ClickHouse automatic deduplication for Replicated tables, not used by SQLMesh
+      tcp_keepalive: [False] # Native client only, specify TCP keepalive configuration. Specify custom keepalive settings as [idle_time_sec, interval_sec, probes], not used by SQLMesh
+      sync_request_timeout: [5] # Timeout for server ping, not used by SQLMesh
+      compress_block_size: [1048576] # Compression block size if compression is enabled, not used by SQLMesh
+    """
+
+    host: str = "localhost"
+    user: str = Field(default="default", alias="username")
+    password: str = ""
+    port: t.Optional[int] = None
+    cluster: t.Optional[str] = None
+    schema_: str = Field(default="default", alias="schema")
+    connect_timeout: int = 10
+    send_receive_timeout: int = 300
+    verify: bool = True
+    compression: str = ""
+    custom_settings: t.Optional[t.Dict[str, t.Any]] = None
+
+    # Not used by SQLMesh
+    driver: t.Optional[str] = None
+    secure: bool = False
+    retries: int = 1
+    database_engine: t.Optional[str] = None
+    cluster_mode: bool = False
+    sync_request_timeout: int = 5
+    compress_block_size: int = 1048576
+    check_exchange: bool = True
+    use_lw_deletes: bool = False
+    allow_automatic_deduplication: bool = False
+    tcp_keepalive: t.Union[bool, t.Tuple[int, ...], t.List[int]] = False
+    database: str = ""
+    local_suffix: str = "local"
+    local_db_prefix: str = ""
+
+    type: Literal["clickhouse"] = "clickhouse"
+
+    def default_incremental_strategy(self, kind: IncrementalKind) -> str:
+        return "legacy"
```

Comment:
What's `legacy`?

Reply:
It's the dbt-clickhouse version of temp table swap.

I wasn't sure whether this method should mimic what dbt-clickhouse would return or our sqlmesh interpretation/version.

Reply:
I see, thanks

Reply:
I don't see us handling it in the conversion code. Is it expected?